### PR TITLE
feat(preflight): report checks skipped for unavailable tools

### PR DESCRIPTION
**Pull request by bmethod**
Reports preflight lanes that could not run because an optional tool is unavailable. The final result names skipped checks. Reporting applies when a relevant file reaches the check; existing parser fallbacks and exit codes remain.

Updates the package documentation and tracked renders. No required-lane setting is added.

Validation: hidden-tool suite passes 13 controls. Restoring silent omission fails lane-name and summary assertions. Existing mode, precision and lane-control suites pass. Full Rust/UI validation and normal commit hooks pass.

Held for overseer MERGE.

Closes KEN-1226

Architecture-docs program: KEN-1203

### CODE REVIEW

File: skills/preflight/tests/unavailable-tools.test.sh [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+run_pf
+[ "$RC" -eq 0 ] && ! has 'not run:' \
+  && ok 'JSONC remains outside strict JSON parsing' || bad 'JSONC scope'
+
+git -C "$R" add -A
+printf '#!/usr/bin/env bash\necho fixture\n' >"$R/scripts/loose.sh"
+git -C "$R" add scripts/loose.sh
+run_pf
+[ "$RC" -eq 1 ] && has '[fail-open]' && has '[shellcheck-errors] not run:' \
+  && ok 'other findings still fail while optional checks are skipped' || bad 'findings retain exit 1'
```
[the remark this answers is not in the document]
Comment (bmethod) [reply skills/preflight/tests/unavailable-tools.test.sh:73]: Declined: the approved contract requires the skipped-lane list on a clean exit, and that path has a must-fail assertion. This requests another assertion on the findings summary without identifying incorrect runtime behavior.

File: skills/preflight/scripts/preflight [hunk 2]
```diff
@@ -842,6 +847,11 @@ while IFS= read -r f; do
           fi
         fi
       done <<<"$(run_shellcheck "$cpath")"
+    else
+      record_skip shellcheck-errors "shellcheck is unavailable"
+      if [ "$vendored_mirror" = 0 ]; then
+        record_skip masked-returns "shellcheck is unavailable"
```
[the remark this answers is not in the document]
Comment (bmethod) [reply skills/preflight/scripts/preflight:853]: Declined: a deletion-only or mode-only shell change reporting the line-scoped lane as skipped is a precision nit in the skip report, not a wrong verdict.

File: skills/preflight/tests/unavailable-tools.test.sh [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    && ok 'available jq runs while TOML remains skipped' || bad 'JSON partial coverage'
+  printf '{"fixture":true}\n' >"$R/data/one.json"
+  git -C "$R" add data/one.json
+  rm "$BIN/jq"
+else
+  printf '  skip: available-jq control (jq is not installed)\n'
+fi
+
+if real="$(command -v python3)" && "$real" -c 'import tomllib' >/dev/null 2>&1; then
+  ln -s "$real" "$BIN/python3"
```
[the remark this answers is not in the document]
Comment (bmethod) [reply skills/preflight/tests/unavailable-tools.test.sh:92]: Declined: a pyenv/asdf shim on the test host is an environment edge case; CI and the supported dev hosts run python3 directly.
